app/uk/gov/hmrc/teamsandrepositories/RepositoryDataSource.scala

```diff
@@ -207,16 +208,18 @@ class CachingRepositoryDataSource[T](
   }
 
   private def dataUpdate() {
-    fromSource.onComplete {
-      case Failure(e) => Logger.warn(s"failed to get latest data due to ${e.getMessage}", e)
-      case Success(d) => {
-        synchronized {
-          this.cachedData = Some(d)
-          Logger.info(s"data update completed successfully")
-
-          if (!initialPromise.isCompleted) {
-            Logger.debug("early clients being sent result")
-            this.initialPromise.success(d)
+    if (enabled) {
```

Comment:
Could we change this a little? I don't think the datasource should be aware of whether or not it is enabled... it would be better from a coupling/encapsulation point of view to instead provide a 'FileDataSource' class that we use in place of this one.